bia-test-data/bia_test_data/data_to_api.py

```
@@ -41,12 +41,176 @@ def get_object_creation_client(
     return private_api
 
 
-def add_objects_to_api(api_base_url, object_list=list[dict]):
+def calculate_dependency_chain_length(
```

I don't think doing it this way is bad on principle, but I'm thinking of when we'll change the models eventually. Suppose we add a new model type that would have dependencies, is there any way that code that uses this would silently ignore the new model (instead of e.g. just failing so we'd know we need to change this too)?

---

So we could make this document based rather than model based. This would mean that all files we sent went through the same, more complicated, system. Perhaps that would be better (would mean there is only ever 1 case to debug, but at least the debugging is always on the same system). It would probably follow a similar 'dependency chain length calculation'

If we added new models and added tests (since this is part of the test package) then the objects wouldn't make it to the API, so the tests would fail if they needed objects in the API. Which feels safe enough to me.

---

Let me know if you think it would be worth changing it to this!